plugins/out_s3/s3.c

```diff
@@ -1114,29 +1149,41 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
             flb_free(payload_buf);
         }
-        m_upload->upload_errors += 1;
-        /* re-add chunk to list */
         if (chunk) {
             s3_store_file_unlock(chunk);
```

Comment:
remove unlock from s3_store.c, its not needed and shouldn't be a function anymore